drivers/sensor/sunxi-thermal.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <debug.h>
+#include <error.h>
+#include <limits.h>
+#include <mmio.h>
+#include <sensor.h>
+#include <sensor/sunxi-thermal.h>
+
+#define SENSOR_COUNT        3
+
+#define THS_0_DATA_REGISTER 0x80
+#define THS_1_DATA_REGISTER 0x84
+#define THS_2_DATA_REGISTER 0x88
+
+enum {
+	THS0,
+	THS1,
+	THS2,
+};
+
+struct sensor_info sunxi_thermal_sensor_info[SENSOR_COUNT] = {
+	[THS0] = {
+		.name    = "ths0",
+		.reading = UINT32_MAX, /* Symbolizes no reading yet. */
+		.address = 0x80,
+	},
+	[THS1] = {
+		.name    = "ths1",
+		.reading = UINT32_MAX, /* Symbolizes no reading yet. */
+		.address = 0x84,
+	},
+	[THS2] = {
+		.name    = "ths2",
+		.reading = UINT32_MAX, /* Symbolizes no reading yet. */
+		.address = 0x88,
+	},
+};
+
+static uint8_t
+sunxi_thermal_get_count(struct device *dev __unused)
```

Comment:
`get_count` hooks are going away since #134. Please update `dev->subdev_count` in the probe function.